### PR TITLE
test(autofix): Full autofix integration task tests w/ celery

### PR DESCRIPTION
**Pull request by jennmueng**
Adds back in real full integration task tests for autofix using VCRs. Also as a bonus, fixes repo client tests that have been plaguing us for a while. 

Right now has 3 tests, root cause, full, and pr creation.

### Next Steps:
- More extensive test cases that test different scenarios and edge cases for an autofix run

### CODE REVIEW

File: tests/automation/autofix/test_autofix_tasks.py [hunk 1]
```diff
@@ -1,175 +1,229 @@
...
+def test_no_state_mapping():
+    state = next(generate(AutofixContinuation))
+    with Session() as session:
+        session.add(DbRunState(id=1, group_id=1, value=state.model_dump(mode="json")))
+        session.flush()
+        session.add(DbPrIdToAutofixRunIdMapping(provider="test", pr_id=1, run_id=1))
+        session.commit()
+
+    retrieved_state = get_autofix_state_from_pr_id("test", 2)
+    assert retrieved_state is None
```
Comment (jennmueng) [inline tests/automation/autofix/test_autofix_tasks.py:129]: This stuff is just removing the class to make the whole file consistent

File: .env.example [hunk 2]
```diff
@@ -5,8 +5,8 @@ LANGFUSE_PUBLIC_KEY=...
 LANGFUSE_HOST=...
 
 ## GitHub
-GITHUB_PRIVATE_KEY=... # Or use GITHUB_TOKEN
-GITHUB_APP_ID=... # Or use GITHUB_TOKEN
+# GITHUB_PRIVATE_KEY=... # Or use GITHUB_TOKEN
+# GITHUB_APP_ID=... # Or use GITHUB_TOKEN
 GITHUB_TOKEN=... # Or use GIHUB_PRIVATE_KEY and GITHUB_APP_ID
```
Comment (jennmueng) [inline .env.example:10]: this is because in any VCR using these we were actually passing ... which is an invalid jwt, we should just comment them out except the token which doesn't do any pre-verification on the gh sdk side

File: src/seer/automation/codebase/repo_client.py [hunk 3]
```diff
@@ -372,10 +371,17 @@ def create_branch_from_changes(
...
+
+        try:
+            branch_ref = self._create_branch(new_branch_name)
+        except GithubException as e:
+            # only use the random suffix if the branch already exists
+            if e.status == 409:
+                new_branch_name = f"{new_branch_name}/{generate_random_string(n=6)}"
+                branch_ref = self._create_branch(new_branch_name)
+            else:
+                raise e
```
Comment (jennmueng) [inline src/seer/automation/codebase/repo_client.py:384]: Branch names were random and new each time, this ensures they're consistent for testing

File: tests/automation/codebase/test_repo_client.py [hunk 4]
```diff
@@ -15,6 +17,14 @@ def clear_repo_client_cache():
...
 
 
+@pytest.fixture(autouse=True)
+def setup_github_app_ids():
+    app_config = resolve(AppConfig)
+    app_config.GITHUB_APP_ID = "123"
+    app_config.GITHUB_PRIVATE_KEY = "456"
+    yield
+
+
```
Comment (jennmueng) [inline tests/automation/codebase/test_repo_client.py:27]: this little change should fix all repo client failures locally

**Review by roaga — APPROVED** [review]
Really nice 🚀 

Though is this approach going to allow us to write tests for rethinking, short-circuiting, Q&A, etc. easily?

**Comment by jennmueng** [comment]
> Really nice 🚀
> 
> Though is this approach going to allow us to write tests for rethinking, short-circuiting, Q&A, etc. easily?

Yes, I have a follow up PR that I've been working on that already has Q&A, and it shouldn't be hard to do the rest as well.
